File: go/algorithm/bubble_sort/bubble_sort.go
package bubble_sort

func Sort(vec *[]int) {
	var swapped bool

	for i := range *vec {
		swapped = false

		for j := 0; j < len(*vec)-i-1; j++ {
			if (*vec)[j] > (*vec)[j+1] {
				(*vec)[j], (*vec)[j+1] = (*vec)[j+1], (*vec)[j]
				swapped = true
			}
		}
		if !swapped {
			break
		}
	}
}

File: js/algorithm/bubble-sort.js
let swapped = true;

function bubbleSort(items) {
  while (swapped) {
    swapped = false;

    for (let i = 0; i < items.length - 1; i++) {
      if (items[i] > items[i + 1]) {
        [items[i], items[i + 1]] = [items[i + 1], items[i]];
        swapped = true;
      }
    }
  }
}

const items = [0, 5, -2, 10, 3, 5];
bubbleSort(items);
console.log(items);
